File: backend/db/model/step_definition.py
from dataclasses import dataclass, field


@dataclass
class StepInput:
    name: str
    type: str
    required: bool
    description: str


@dataclass
class StepDefinition:
    name: str
    id: str
    version: str
    category: str
    description: str
    inputs: list[StepInput]
    required_permissions: dict[str, set[str]] = field(default_factory=dict)

File: frontend/src/types/stepDefinition.ts
type StepDefinition = {
  name: string;
  id: string;
  version: string;
  category: string;
  description: string;
  inputs: StepInput[];
  required_permissions: Record<string, string>[];
};

type StepInput = {
  name: string;
  type: string;
  required: boolean;
  description: string;
};
